src/core/hle/kernel/k_writable_event.h

```diff
@@ -4,36 +4,32 @@
 
 #pragma once
 
-#include "core/hle/kernel/object.h"
+#include "core/hle/kernel/k_auto_object.h"
+#include "core/hle/kernel/slab_helpers.h"
 #include "core/hle/result.h"
 
 namespace Kernel {
 
 class KernelCore;
 class KEvent;
 
-class KWritableEvent final : public Object {
+class KWritableEvent final
+    : public KAutoObjectWithSlabHeapAndContainer<KWritableEvent, KAutoObjectWithList> {
+    KERNEL_AUTOOBJECT_TRAITS(KWritableEvent, KAutoObject);
+
 public:
-    explicit KWritableEvent(KernelCore& kernel, std::string&& name);
+    explicit KWritableEvent(KernelCore& kernel);
     ~KWritableEvent() override;
 
-    std::string GetTypeName() const override {
-        return "KWritableEvent";
-    }
-
-    static constexpr HandleType HANDLE_TYPE = HandleType::WritableEvent;
-    HandleType GetHandleType() const override {
-        return HANDLE_TYPE;
-    }
-
-    void Initialize(KEvent* parent_);
+    virtual void Destroy() override;
 
-    void Finalize() override {}
+    static void PostDestroy([[maybe_unused]] uintptr_t arg) {}
 
+    void Initialize(KEvent* parent_, std::string&& name_);
     ResultCode Signal();
     ResultCode Clear();
 
-    KEvent* GetParent() const {
+    constexpr KEvent* GetParent() const {
```

Comment:
```suggestion
    KEvent* GetParent() const {
```

This can't be used in a constexpr context